### DISCUSSION TITLE
Does the results list have a maximum length (number of items?)

### DISCUSSION THREAD

**Discussion by RedBearAK** [Ulauncher/Ulauncher#941] [General]
The results list for Ulauncher has a shortcut next to each item. On a 1080p laptop screen, I've seen up to `Alt+g` being displayed, although right now it's only showing results up to `Alt+f` for some reason, no matter what search I try. Does the results list have a set maximum length, or would it show even more items on a larger/higher resolution display? 

Haven't seen anything in the preferences related to this other than "number of recent apps". 

Need to know the hard-coded limit, if there is one, for the maximum number of items that will ever be shown on the results list.

**Comment by friday** [comment]
Yes. App search mode is hard coded to 9, and file browser mode that uses the compact result view is hard coded to 17. Since there are two different view modes we can't make it a generic setting that covers both. I think it's better to properly detect the height, but I haven't had the time to look at that.

**Reply by RedBearAK** [reply]
> Yes. App search mode is hard coded to 9, and file browser mode that uses the compact result view is hard coded to 17. Since there are two different view modes we can't make it a generic setting that covers both. I think it's better to properly detect the height, but I haven't had the time to look at that.

@friday 

So just to be clear, in the future you may be planning to allow the file browser mode list to extend beyond 17 items? Potentially all the way to the end of the alphabet? But right now the maximum that should be shown is up to `Alt+h` (items 1-9 plus abcdefgh). Is this correct?

**Reply by friday** [reply]
> So just to be clear, in the future you may be planning to allow the file browser mode list to extend beyond 17 items? Potentially all the way to the end of the alphabet? But right now the maximum that should be shown is up to `Alt+h` (items 1-9 plus abcdefgh). Is this correct?

That is the logical outcome of removing this hard limit. You are welcome to create a pull request.

**Reply by RedBearAK** [reply]
Yeah, I'm not at that skill level. But I'd like to present the idea that allowing the list to expand would not necessarily be good. It would create a potentially very confusing long list, up to 35 items long. Albert launcher limits itself to 10 visible items and then becomes scrollable, and lets the user choose whether to always show the scroll bar or only when the list overflows the max visible items. Letting the list expand based on screen size would create a different experience for different screens. 

There is an "incremental sort" option to make a long scrollable list of results appear faster, since an issue of how to order the list crops up when only partial results are available. 

If you continue to limit the results to 17 and then allow the list to be scrollable, you can also limit the number of keyboard shortcuts by simply keeping each shortcut attached to the lines that are visible while scrolling. As in, `Alt+1` always activates the first visible result, rather than being attached to a specific result. That would mean you'd never need more than up to `Alt+h` for keyboard shortcuts. 

I can push this into a feature request issue if you think any of this makes sense.
